core/src/main/java/org/apache/iceberg/encryption/DefaultEncryptionManagerFactory.java

```diff
@@ -0,0 +1,128 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.encryption;
+
+import static org.apache.iceberg.TableProperties.DEFAULT_FILE_FORMAT;
+import static org.apache.iceberg.TableProperties.ENCRYPTION_TABLE_KEY;
+
+import java.io.IOException;
+import java.util.Map;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableProperties;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.util.PropertyUtil;
+
+public class DefaultEncryptionManagerFactory implements EncryptionManagerFactory {
+  public static final String ICEBERG_ENCRYPTION_TABLE_KEY = "iceberg.encryption.table.key.id";
+
+  private static class KmsClientSupplier {
+    private final boolean ignoreTableProperties;
+
+    KmsClientSupplier(boolean ignoreTableProperties) {
+      this.ignoreTableProperties = ignoreTableProperties;
+    }
+
+    KeyManagementClient create(Map<String, String> encryptionProperties) {
+      // load kms impl from catalog properties, if not present fall back to table properties (if
+      // allowed).
+      String kmsImpl = encryptionProperties.get(CatalogProperties.ENCRYPTION_KMS_CLIENT_IMPL);
+
+      Preconditions.checkArgument(
+          null != kmsImpl,
+          "KMS Client implementation class is not set (via "
+              + CatalogProperties.ENCRYPTION_KMS_CLIENT_IMPL
+              + " catalog property nor "
+              + TableProperties.ENCRYPTION_KMS_CLIENT_IMPL
+              + " table property). "
+              + "IgnoreTableProperties is: "
+              + ignoreTableProperties);
+
+      return KmsUtil.loadKmsClient(kmsImpl, encryptionProperties);
+    }
+  }
+
+  private KmsClientSupplier kmsClientSupplier;
+  private KeyManagementClient client;
+  private Map<String, String> catalogPropertyMap;
+  private boolean ignoreTableProperties;
+
+  @Override
+  public void initialize(Map<String, String> catalogProperties) {
+    this.catalogPropertyMap = catalogProperties;
+    ignoreTableProperties =
+        PropertyUtil.propertyAsBoolean(
+            catalogProperties,
+            CatalogProperties.ENCRYPTION_IGNORE_TABLE_PROPS,
+            CatalogProperties.ENCRYPTION_IGNORE_TABLE_PROPS_DEFAULT);
+    kmsClientSupplier = new KmsClientSupplier(ignoreTableProperties);
+  }
+
+  @Override
+  public EncryptionManager create(TableMetadata tableMetadata) {
+    if (null == tableMetadata) {
+      return PlaintextEncryptionManager.INSTANCE;
+    }
+
+    Map<String, String> tableProperties = tableMetadata.properties();
+    String fileFormat =
+        PropertyUtil.propertyAsString(tableProperties, DEFAULT_FILE_FORMAT, "parquet");
+    if (!fileFormat.equals("parquet")) {
```

Comment:
`FileFormat.fromString` should be used to match the general usage pattern.
In other cases we accept upper case `PARQUET` too, so I think we should accept it here too.

Reply:
SGTM, thanks for the suggestion.